docs/src/appendix/chisel-plugin.md

```diff
@@ -0,0 +1,76 @@
+# Notes on the Compiler Plug-in
+
+The Chisel plugin provides some operations that are too difficult, or not possbile, 
+to implement through regular Scala code.
+
+## Compiler plugin operations
+1. Automatically generates the `cloneType` methods of Bundles
+2. Records that this plugin ran on a bundle by adding a method `_usingPlugin: Boolean = true`
+3. Generates a function that returns list of the hardware fields in a bundle `_elementsImpl`
+
+### Generating `cloneType` method
+After a period of testing this is now on by default. This flag was used to control that prior to full adoption.
+```
+-P:chiselplugin:useBundlePlugin
+```
+
+### Generating the `_elementsImpl` method
+For generating the elementsImpl method, the default is off.
+Turn it on by using the javac flag
+```
+-P:chiselplugin:buildElementAccessor
+```
+
+#### Detecting Bundles with Seq[Data]
```

Comment:
I'm confused what this section is saying. does the compiler plugin detect this or not? What does "currently" mean?

Reply:
I've cleaned up the language a bit here. Basically says this feature (3. on earlier list) is off by default and uses a scalac flag to turn it on.